java/java.lsp.server/vscode/src/extension.ts

```diff
@@ -675,6 +676,53 @@ function doActivateWithJDK(specifiedJDK: string | null, context: ExtensionContex
         window.showErrorMessage('Error initializing ' + reason);
     });
 
+    async function showHtmlPage(params : HtmlPageParams): Promise<string> {
```

Comment:
When there is a need to show the UI, the JVM opens a port (unsecured right now) with a simplistic HTTP server ands sends a LSP request to show a page where it passes the URL to show. The request is handled in this VSCode extension and opens a `WebView` that connects to the HTTP port and makes the page "live" and controllable from a Java code.